src/lib.ts

```diff
@@ -0,0 +1,320 @@
+import sharp from 'sharp';
+import smartcrop from 'smartcrop-sharp';
+import imageminPngquant from 'imagemin-pngquant';
+import { S3Client, S3ClientConfig, GetObjectCommand, GetObjectCommandOutput } from '@aws-sdk/client-s3';
+
+const authenticatedRequest = !!process.env.S3_AUTHENTICATED_REQUEST
+	? process.env.S3_AUTHENTICATED_REQUEST.toLowerCase() == 'true'
+	: false;
+
+export interface Args {
+	webp?: string | boolean;
+	background?: string;
+	lb?: string;
+	zoom?: string;
+	crop?: string | string[];
+	fit?: string;
+	gravity?: string;
+	crop_strategy?: string;
+	resize?: string | number[];
+	quality?: string | number;
+	h?: string;
+	w?: string;
+	key: string;
```

Comment:
Easy to miss this is required, suggest moving to top so required before optional. Also, alphabetical?